src/component/chat.cpp

```diff
@@ -101,6 +103,66 @@ namespace chat
 				g_say_hook.invoke<void>(ent, target, mode, chatText);
 			}
 		}
+
+		std::map<std::string, std::string> colors_users;
+
+		// exemple:
+		// { 
+		//	 "purple": ["DoktorSAS"],
+		//	 "red" : ["fed"]
+		// }
+
+		void load_colors_data()
+		{
+			if (!utils::io::file_exists("colors/chat.json"))
+			{
+				printf("chat.json was not found\n");
+				return;
+			}
+
+
+			nlohmann::json obj;
+			try
+			{
+				obj = nlohmann::json::parse(utils::io::read_file("colors/chat.json").data());
+			}
+			catch (nlohmann::json::parse_error&)
+			{
+				printf("json parse error\n");
+				return;
+			}
+
+			std::vector<std::string> keys = { "black", "red", "green", "yellow", "blue", "cyan", "purple", "white", "grey", "brown" };
+			std::vector<std::string> values = { "^0", "^1", "^2", "^3", "^4", "^5", "^6", "^7", "^8", "^9" };
+
+			for (int i = 0; i < keys.size(); i++)
+			{
+				if (obj.contains(std::string{ keys.at(i) }))
+				{
+					std::vector<std::string> current = obj[keys.at(i)];
+					for (int j = 0; j < current.size(); j++)
+					{
+						colors_users[current.at(j).c_str()] = values.at(i);
+					}
+				}
+			}
+		}
+
+		const char* g_say_va_hook(const char* msg, int mode, const char* teamColor, const char* name, const char* szStateString, const char* teamString, int a6, const char* color, const char* message)
```

Comment:
The hook is wrong.
"%c \"\x14GAME_CHAT_MESSAGE\x15%s\x15%s\x14%s\x14%s\x15%c%c%s\"" the second argument is a char, second to last is also a char.
Snake case function arguments.

Reply:
I did not get it.